File: okex-java-sdk-api-v5/src/main/java/com/okex/open/api/service/account/impl/AccountAPI.java
package com.okex.open.api.service.account.impl;

import com.alibaba.fastjson.JSONObject;
import com.okex.open.api.bean.account.param.AccountQuickBorrowRepay;
import retrofit2.Call;
import retrofit2.http.*;


public interface AccountAPI {

    //查看账户余额 Get Balance
    @GET("/api/v5/account/balance")
    Call<JSONObject> getBalance(@Query("ccy") String ccy);

    //查看持仓信息 Get Positions
    @GET("/api/v5/account/positions")
    Call<JSONObject> getPositions(@Query("instType") String instType,@Query("instId") String instId,@Query("posId") String posId);

    //查看历史持仓信息   Get positions-history
    @GET("/api/v5/account/positions-history")
    Call<JSONObject> getPositionsHistory(@Query("instType") String instType, @Query("instId") String instId, @Query("mgnMode") String mgnMode, @Query("type") String type, @Query("after") String after, @Query("before") String before, @Query("limit") String limit,@Query("posId") String posId);

    //查看账户持仓风险 Get account and position risk
    @GET("/api/v5/account/account-position-risk")
    Call<JSONObject> getAccountAndPosition(@Query("instType") String instType);

    //账单流水查询（近七天） Get Bills Details (last 7 days)
    @GET("/api/v5/account/bills")
    Call<JSONObject> getBillsDetails7Days(@Query("instType") String instType,
                                          @Query("ccy")String ccy,
                                          @Query("mgnMode")String mgnMode,
                                          @Query("ctType")String ctType,
                                          @Query("type")String type,
                                          @Query("subType")String subType,
                                          @Query("after")String after,
                                          @Query("before")String before,
                                          @Query("limit")String limit);

    //账单流水查询（近三月） Get Bills Details (last 3 months)
    @GET("/api/v5/account/bills-archive")
    Call<JSONObject> getBillsDetails3Months(@Query("instType") String instType,
                                          @Query("ccy")String ccy,
                                          @Query("mgnMode")String mgnMode,
                                          @Query("ctType")String ctType,
                                          @Query("type")String type,
                                          @Query("subType")String subType,
                                          @Query("after")String after,
                                          @Query("before")String before,
                                          @Query("limit")String limit);

    //查看账户配置 Get Account Configuration
    @GET("/api/v5/account/config")
    Call<JSONObject> getAccountConfiguration();

    //设置持仓模式 Set Position mode
    @POST("/api/v5/account/set-position-mode")
    Call<JSONObject> setPositionMode(@Body JSONObject jsonObject);

    //设置杠杆倍数 Set Leverage
    @POST("/api/v5/account/set-leverage")
    Call<JSONObject> setLeverage(@Body JSONObject jsonObject);

    //获取最大可买卖/开仓数量 Get maximum buy/sell amount or open amount
    @GET("/api/v5/account/max-size")
    Call<JSONObject> getMaximumTradableSizeForInstrument(@Query("instId") String instId,@Query("tdMode") String tdMode,@Query("ccy") String ccy,@Query("px") String px,@Query("leverage") String leverage,@Query("unSpotOffset") Boolean unSpotOffset);

    //获取最大可用数量 Get Maximum Tradable Size For Instrument
    @GET("/api/v5/account/max-avail-size")
    Call<JSONObject> getMaximumAvailableTradableAmount(@Query("instId") String instId,@Query("tdMode") String tdMode,@Query("ccy") String ccy,@Query("reduceOnly") Boolean reduceOnly,@Query("px") String px,@Query("unSpotOffset") Boolean unSpotOffset,@Query("quickMgnType") String quickMgnType);

    //调整保证金 Increase/Decrease margin
    @POST("/api/v5/account/position/margin-balance")
    Call<JSONObject> increaseDecreaseMargin(@Body JSONObject jsonObject);

    //获取杠杆倍数 Get Leverage
    @GET("/api/v5/account/leverage-info")
    Call<JSONObject> getLeverage(@Query("instId") String instId,@Query("mgnMode") String mgnMode);

    //获取交易产品最大可借 Get the maximum loan of instrument
    @GET("/api/v5/account/max-loan")
    Call<JSONObject> getTheMaximumLoanOfIsolatedMARGIN(@Query("instId")String instId,@Query("mgnMode")String mgnMode,@Query("mgnCcy")String mgnCcy);

    //获取当前账户交易手续费费率 Get Fee Rates
    @GET("/api/v5/account/trade-fee")
    Call<JSONObject> getFeeRates(@Query("instType")String instType,@Query("instId")String instId,@Query("uly")String uly,@Query("instFamily")String instFamily);

    //获取计息记录 Get interest-accrued
    @GET("/api/v5/account/interest-accrued")
    Call<JSONObject> getInterestAccrued(@Query("type")String type,@Query("instId")String instId,@Query("ccy")String ccy,@Query("mgnMode")String mgnMode,@Query("after")String after,@Query("before")String before,@Query("limit")String limit);

    //获取用户当前杠杆借币利率 Get interest rate
    @GET("/api/v5/account/interest-rate")
    Call<JSONObject> getInterestRate(@Query("ccy")String ccy);

    //期权希腊字母PA/BS切换 Set the display type of Greeks
    @POST("/api/v5/account/set-greeks")
    Call<JSONObject> setTheDisplayTypeOfGreeks(@Body JSONObject jsonObject);

    //逐仓交易设置 Isolated margin trading settings
    @POST("/api/v5/account/set-isolated-mode")
    Call<JSONObject> setIsolatedMode(@Body JSONObject jsonObject);

    //查看账户最大可转余额 Get Maximum Withdrawals
    @GET("/api/v5/account/max-withdrawal")
    Call<JSONObject> getMaximumWithdrawals(@Query("ccy") String ccy);

    //查看账户特定风险状态 Get account risk state
    @GET("/api/v5/account/risk-state")
    Call<JSONObject> getRiskState();

    //一键借币模式手动借币还币 Manual borrow and repay in Quick Margin Mode
    @POST("/api/v5/account/quick-margin-borrow-repay")
    Call<JSONObject> quickMarginBorrowRepay(@Body JSONObject jsonObject);

    //获取一键借币还币历史 Get manual borrow and repay history in Quick Margin Mode
    @GET("/api/v5/account/borrow-repay-history")
    Call<JSONObject> getMarginBorrowRepayHistory(@Query("instId") String instId,@Query("ccy")  String ccy,@Query("side")  String side,@Query("after")  String after,@Query("before")  String before,@Query("begin")  String begin,@Query("end")  String end,@Query("limit")  String limit);

    //尊享借币还币  VIP loans borrow and repay
    @POST("/api/v5/account/borrow-repay")
    Call<JSONObject> borrowRepay(@Body JSONObject parseObject);

    //获取尊享借币借还历史  Get borrow and repay history for VIP loans
    @GET("/api/v5/account/borrow-repay-history")
    Call<JSONObject> getBorrowRepayHistory(@Query("ccy") String ccy,@Query("after") String after,@Query("before") String before,@Query("limit") String limit);

    //获取尊享借币计息记录 Get VIP interest accrued data
    @GET("/api/v5/account/vip-interest-accrued")
    Call<JSONObject>  getVipInterestAccrued(@Query("ccy") String ccy,@Query("ordId")  String ordId,@Query("after")  String after,@Query("before")  String before,@Query("limit")  String limit);

    //获取尊享借币扣息记录 Get VIP interest deducted data
    @GET("/api/v5/account/vip-interest-deducted")
    Call<JSONObject>  getVipInterestDeducted(@Query("ordId") String ordId,@Query("ccy") String ccy,@Query("after") String after,@Query("before") String before,@Query("limit") String limit);

    //尊享借币订单列表 Get VIP loan order list
    @GET("/api/v5/account/vip-loan-order-list")
    Call<JSONObject>  getVipLoanOrderList(@Query("ordId") String ordId, @Query("state") String state, @Query("ccy") String ccy, @Query("after") String after, @Query("before") String before, @Query("limit") String limit);

    //尊享借币订单详情 Get VIP loan order detail
    @GET("/api/v5/account/vip-loan-order-detail")
    Call<JSONObject>  getVipLoanOrderDetail(@Query("ccy") String ccy,@Query("ordId")  String ordId,@Query("after")  String after,@Query("before")  String before,@Query("limit")  String limit);

    //获取借币利率与限额  Get borrow interest and limit
    @GET("/api/v5/account/interest-limits")
    Call<JSONObject> getInterestLimits(@Query("type") String type,@Query("ccy") String ccy);

    //组合保证金的虚拟持仓保证金计算 Position builder
    @POST("/api/v5/account/simulated_margin")
    Call<JSONObject> simulatedMargin(@Body JSONObject parseObject);

    //查看账户Greeks Get Greeks
    @GET("/api/v5/account/greeks")
    Call<JSONObject> getAccountGreeks(@Query("ccy") String ccy);

    //获取组合保证金模式全仓限制 Get PM limitation
    @GET("/api/v5/account/position-tiers")
    Call<JSONObject> getPositionTiers(@Query("instType") String instType,@Query("uly")  String uly,@Query("instFamily")  String instFamily);

    // 设置组合保证金账户风险对冲模式 Set risk offset type
    @POST("/api/v5/account/set-riskOffset-type")
    Call<JSONObject> setRiskOffsetType(@Body JSONObject parseObject);

    //开通期权交易 Activate option
    @POST("/api/v5/account/activate-option")
    Call<JSONObject> activateOption();
}

File: okx-python-sdk-api-v5/okx/Account_api.py
from .client import Client
from .consts import *


class AccountAPI(Client):

    def __init__(self, api_key, api_secret_key, passphrase, use_server_time=False, flag='1'):
        Client.__init__(self, api_key, api_secret_key, passphrase, use_server_time, flag)

    # Get Positions
    def get_position_risk(self, instType=''):
        params = {}
        if instType:
            params['instType'] = instType
        return self._request_with_params(GET, POSITION_RISK, params)

    # Get Balance
    def get_account(self, ccy=''):
        params = {}
        if ccy:
            params['ccy'] = ccy
        return self._request_with_params(GET, ACCOUNT_INFO, params)

    # Get Positions
    def get_positions(self, instType='', instId=''):
        params = {'instType': instType, 'instId': instId}
        return self._request_with_params(GET, POSITION_INFO, params)

    # Get Bills Details (recent 7 days)
    def get_bills_detail(self, instType='', instId='', ccy='', mgnMode='', ctType='', type='', subType='', after='', before='',begin='',end='',
                         limit=''):
        params = {'instType': instType, 'ccy': ccy, 'mgnMode': mgnMode, 'ctType': ctType, 'type': type,
                  'subType': subType, 'after': after, 'before': before, 'limit': limit, 'instId':instId, 'begin':begin, 'end':end}
        return self._request_with_params(GET, BILLS_DETAIL, params)

    # Get Bills Details (recent 3 months)
    def get_bills_details(self, instType='', instId = '', ccy='', mgnMode='', ctType='', type='', subType='', after='', before='',begin='',end='',
                          limit=''):
        params = {'instType': instType, 'ccy': ccy, 'mgnMode': mgnMode, 'ctType': ctType, 'type': type,
                  'subType': subType, 'after': after, 'before': before, 'limit': limit, 'instId':instId, 'begin':begin, 'end':end}
        return self._request_with_params(GET, BILLS_ARCHIVE, params)

    # Get Account Configuration
    def get_account_config(self):
        return self._request_without_params(GET, ACCOUNT_CONFIG)

    # Get Account Configuration
    def get_position_mode(self, posMode):
        params = {'posMode': posMode}
        return self._request_with_params(POST, POSITION_MODE, params)

    # Get Account Configuration
    def set_leverage(self, lever, mgnMode, instId='', ccy='', posSide=''):
        params = {'lever': lever, 'mgnMode': mgnMode, 'instId': instId, 'ccy': ccy, 'posSide': posSide}
        return self._request_with_params(POST, SET_LEVERAGE, params)

    # Get Maximum Tradable Size For Instrument
    def get_maximum_trade_size(self, instId, tdMode, ccy='', px='', leverage='',unSpotOffset=''):
        params = {'instId': instId, 'tdMode': tdMode, 'ccy': ccy, 'px': px, 'leverage':leverage,'unSpotOffset':unSpotOffset}
        return self._request_with_params(GET, MAX_TRADE_SIZE, params)

    # Get Maximum Available Tradable Amount
    def get_max_avail_size(self, instId, tdMode, ccy='', reduceOnly='', unSpotOffset='',quickMgnType='',px=''):
        params = {'instId': instId, 'tdMode': tdMode, 'ccy': ccy, 'reduceOnly': reduceOnly,
                  'unSpotOffset':unSpotOffset,'quickMgnType':quickMgnType,'px': px}
        return self._request_with_params(GET, MAX_AVAIL_SIZE, params)

    # Increase / Decrease margin
    def Adjustment_margin(self, instId, posSide, type, amt,loanTrans=''):
        params = {'instId': instId, 'posSide': posSide, 'type': type, 'amt': amt,'loanTrans':loanTrans}
        return self._request_with_params(POST, ADJUSTMENT_MARGIN, params)

    # Get Leverage
    def get_leverage(self, instId, mgnMode, ccy):
        params = {'instId': instId, 'mgnMode': mgnMode, 'ccy':ccy}
        return self._request_with_params(GET, GET_LEVERAGE, params)

    # Get the maximum loan of isolated MARGIN
    def get_max_load(self, instId, mgnMode, mgnCcy):
        params = {'instId': instId, 'mgnMode': mgnMode, 'mgnCcy': mgnCcy}
        return self._request_with_params(GET, MAX_LOAN, params)

    # Get Fee Rates
    def get_fee_rates(self, instType = '', instId='', uly='', category='', instFamily='',ruleType = ''):
        params = {'instType': instType, 'instId': instId, 'uly': uly, 'category': category,'instFamily':instFamily,'ruleType':ruleType}
        return self._request_with_params(GET, FEE_RATES, params)

    # Get interest-accrued
    def get_interest_accrued(self, instId='', ccy='', mgnMode='', after='', before='', limit='', type=''):
        params = {'instId': instId, 'ccy': ccy, 'mgnMode': mgnMode, 'after': after, 'before': before, 'limit': limit, 'type':type}
        return self._request_with_params(GET, INTEREST_ACCRUED, params)

    # Get interest-accrued
    def get_interest_rate(self, ccy=''):
        params = {'ccy': ccy}
        return self._request_with_params(GET, INTEREST_RATE, params)

    # Set Greeks (PA/BS)
    def set_greeks(self, greeksType):
        params = {'greeksType': greeksType}
        return self._request_with_params(POST, SET_GREEKS, params)

    # Set Isolated Mode
    def set_isolated_mode(self, isoMode,type):
        params = {'isoMode': isoMode, 'type':type}
        return self._request_with_params(POST, ISOLATED_MODE, params)

    # Get Maximum Withdrawals
    def get_max_withdrawal(self, ccy=''):
        params = {'ccy': ccy}
        return self._request_with_params(GET, MAX_WITHDRAWAL, params)

    # Get account risk state
    def get_account_risk(self):
        return self._request_without_params(GET, ACCOUNT_RISK)

    # Get borrow repay
    def borrow_repay(self, ccy='', side='', amt='', ordId = ''):
        params = {'ccy': ccy, 'side': side, 'amt': amt, 'ordId':ordId}
        return self._request_with_params(POST, BORROW_REPAY, params)

    # Get borrow repay history
    def get_borrow_repay_history(self, ccy='', after='', before='', limit=''):
        params = {'ccy': ccy, 'after': after, 'before': before, 'limit':limit}
        return self._request_with_params(GET, BORROW_REPAY_HISTORY, params)

    # Get Obtain borrowing rate and limit
    def get_interest_limits(self, type='',ccy=''):
        params = {'type': type, 'ccy': ccy}
        return self._request_with_params(GET, INTEREST_LIMITS, params)

    # Get Simulated Margin
    def get_simulated_margin(self, instType	='',inclRealPos='',spotOffsetType='',simPos=[]):
        params = {'instType': instType, 'inclRealPos': inclRealPos,'spotOffsetType':spotOffsetType,'simPos': simPos}
        return self._request_with_params(POST, SIMULATED_MARGIN, params)

    # Get  Greeks
    def get_greeks(self, ccy=''):
        params = {'ccy': ccy,}
        return self._request_with_params(GET, GREEKS, params)

    def get_positions_history(self, instType='', instId = '', mgnMode = '', type = '', after = '', before = '', limit = '', posId = ''):
        params = {'instType': instType, 'instId':instId, 'mgnMode':mgnMode, 'type':type, 'after':after, 'before':before, 'limit':limit, 'posId':posId}
        return self._request_with_params(GET, POSITIONS_HISTORY, params)

    def position_tiers(self, instType='',uly=''):
        params = {'instType': instType, 'uly': uly}
        return self._request_with_params(GET, POSITION_TIRES, params)

    def activate_option(self):
        params = {}
        return self._request_with_params(POST, ACTIVATE_OPTION, params)

    def set_auto_loan(self,autoLoan = ''):
        params = {'autoLoan':autoLoan}
        return self._request_with_params(POST, SET_AUTO_LOAN, params)

    def quick_margin_borrow_repay(self, instId='', ccy='', side='', amt=''):
        params = {'instId':instId, 'ccy':ccy, 'side':side, 'amt':amt}
        return self._request_with_params(POST, QUICK_MARGIN_BRROW_REPAY, params)

    def quick_margin_borrow_repay_history(self, instId='', ccy='', side='', after='', before='', begin='', end='', limit=''):
        params = {'instId': instId, 'ccy': ccy, 'side': side, 'after': after, 'before': before, 'begin': begin, 'end': end, 'limit': limit}
        return self._request_with_params(GET, QUICK_MARGIN_BORROW_REPAY_HISTORY, params)

    # GET /api/v5/account/vip-interest-accrued
    def vip_interest_accrued(self, ccy = '', ordId = '', after = '', before = '', limit = ''):
        params = {'ccy': ccy, 'ordId': ordId, 'after': after, 'before': before, 'limit': limit}
        return self._request_with_params(GET, VIP_INTEREST_ACCRUED, params)

    # GET /api/v5/account/vip-interest-deducted
    def vip_interest_deducted(self, ccy = '', ordId = '', after = '', before = '', limit = ''):
        params = {'ccy': ccy, 'ordId': ordId, 'after': after, 'before': before, 'limit': limit}
        return self._request_with_params(GET, VIP_INTEREST_DEDUCTED, params)

    # GET /api/v5/account/vip-loan-order-list
    def vip_loan_order_list(self, ordId = '', state = '', ccy = '', after = '', before = '', limit = ''):
        params = {'ordId': ordId, 'state': state, 'ccy': ccy, 'after': after, 'before': before, 'limit': limit}
        return self._request_with_params(GET, VIP_LOAN_ORDER_LIST, params)

    # GET /api/v5/account/vip-loan-order-detail
    def vip_loan_order_detail(self, ccy = '', ordId = '', after = '', before = '', limit = ''):
        params = {'ccy': ccy, 'ordId': ordId, 'after': after, 'before': before, 'limit': limit}
        return self._request_with_params(GET, VIP_LOAN_ORDER_DETAIL, params)

    # POST /api/v5/account/subaccount/set-loan-allocation
    def set_loan_allocation(self, enable, alloc:[]):
        params = {'enable': enable, 'alloc': alloc}
        return self._request_with_params(POST, SET_LOAN_ALLOCATION, params)

    # GET /api/v5/account/subaccount/interest-limits
    def interest_limits(self, subAcct = '', ccy = ''):
        params = {'subAcct': subAcct, 'ccy': ccy}
        return self._request_with_params(GET, INTEREST_LIMITS, params)

    # POST /api/v5/account/set-riskOffset-type
    def set_riskOffset_type(self, type = ''):
        params = {'type':type}
        return self._request_with_params(POST, SET_RISKOFFSET_TYPE, params)

    # POST /api/v5/account/mmp-reset
    def mmp_reset(self,instType,instFamily):
        params = {'instType': instType,'instFamily':instFamily}
        return self._request_with_params(POST, MMP_RESET, params)

    # POST /api/v5/account/mmp-config
    def mmp_config(self,instFamily,timeInterval,frozenInterval,qtyLimit):
        params = {'instFamily': instFamily,'timeInterval': timeInterval,'frozenInterval': frozenInterval,'qtyLimit': qtyLimit,}
        return self._request_with_params(POST, MMP_CONFIG, params)

    # GET /api/v5/account/mmp-config
    def mmp(self,instFamily):
        params = {'instFamily': instFamily}
        return self._request_with_params(GET, MMP, params)

    # POST /api/v5/account/set-account-level
    def set_account_level(self,acctLv):
        params = {'acctLv': acctLv}
        return self._request_with_params(POST, SET_ACCOUNT_LEVEL, params)


    def position_builder(self,inclRealPosAndEq='',spotOffsetType='',simPos='',simAsset='',
                         greeksType='',):
        params = {'acctLv': acctLv, 'spotOffsetType': spotOffsetType, 'simPos': simPos, 'simAsset': simAsset,
                  'greeksType': greeksType, }
        return self._request_with_params(POST, POSITION_BUILDER, params)


    # POST /api/v5/account/set-riskOffset-amt
    def set_riskOffset_amt(self,ccy = '', clSpotInUseAmt = ''):
        params = {'ccy': ccy, 'clSpotInUseAmt': clSpotInUseAmt}
        return self._request_with_params(POST, SET_RISKOFFSET_AMT, params)

    # GET /api/v5/account/fixed-loan/borrowing-limit
    def get_fixed_loan_borrowing_limit(self):
        params = {}
        return self._request_with_params(GET, GET_FIXED_LOAN_BORROWING_LIMIT, params)

    # GET /api/v5/account/fixed-loan/borrowing-quote
    def get_fixed_loan_borrowing_quote(self,type = '', ccy = '', amt = '', maxRate = '', term = '', ordId = ''):
        params = {'type':type, 'ccy':ccy, 'amt':amt, 'maxRate':maxRate, 'maxRate':maxRate, 'term':term, 'ordId':ordId}
        return self._request_with_params(GET, GET_FIXED_LOAN_BORROWING_QUOTE, params)


    # POST /api/v5/account/fixed-loan/borrowing-order
    def fixed_loan_borrowing_order(self,ccy = '', amt = '', maxRate = '', term = '', reborrow = '', reborrowRate = ''):
        params = {'ccy':ccy, 'amt':amt, 'maxRate':maxRate, 'term':term, 'reborrow':reborrow, 'reborrowRate':reborrowRate}
        return self._request_with_params(POST, FIXED_LOAN_BORROWING_ORDER, params)


    # POST /api/v5/account/fixed-loan/amend-borrowing-order
    def fixed_loan_amend_borrowing_order(self,ordId = '', reborrow = '', renewMaxRate = ''):
        params = {'ordId':ordId, 'reborrow':reborrow, 'renewMaxRate':renewMaxRate}
        return self._request_with_params(POST, FIXED_LOAN_AMEND_BORROWING_ORDER, params)


    # POST /api/v5/account/fixed-loan/manual-reborrow
    def fixed_loan_manual_reborrow(self,ordId = '', maxRate = ''):
        params = {'ordId':ordId, 'maxRate':maxRate}
        return self._request_with_params(POST, FIXED_LOAN_MANUAL_BORROWING, params)

    # POST /api/v5/account/fixed-loan/repay-borrowing-order
    def fixed_loan_repay_borrowing_order(self,ordId = ''):
        params = {'ordId':ordId}
        return self._request_with_params(POST, FIXED_LOAN_REPAY_BORROWING_ORDER, params)


    # GET /api/v5/account/fixed-loan/borrowing-orders-list
    def get_fixed_loan_borrowing_orders_list(self,ordId = '', ccy = '', state = '', after = '', before = '',limit = '',term = ''):
        params = {'ordId':ordId, 'ccy':ccy, 'state':state, 'after':after, 'before':before, 'limit':limit, 'term':term}
        return self._request_with_params(GET, GET_FIXED_LOAN_BORROWING_ORDERS_LIST, params)

    # GET /api/v5/account/instruments
    def get_account_instruments(self,instType = '', uly = '', instFamily = '', instId = ''):
        params = {'instType':instType, 'uly':uly, 'instFamily':instFamily, 'instId':instId}
        return self._request_with_params(GET, GET_ACCOUNT_INSTRUMENTS, params)


    # POST /api/v5/account/spot-manual-borrow-repay
    def spot_manual_borrow_repay(self,ccy = '', side = '', amt = ''):
        params = {'ccy':ccy, 'side':side, 'amt':amt}
        return self._request_with_params(POST, SPOT_MANUAL_BORROW_REPAY, params)


    # POST /api/v5/account/set-auto-repay
    def set_auto_repay(self,autoRepay = ''):
        params = {'autoRepay':autoRepay}
        return self._request_with_params(POST, SET_AUTO_REPAY, params)


    # GET /api/v5/account/spot-borrow-repay-history
    def get_spot_borrow_repay_history(self,ccy = '', type = '', after = '', before = '', limit = ''):
        params = {'ccy':ccy, 'type':type, 'after':after, 'before':before, 'limit':limit}
        return self._request_with_params(GET, GET_SPOT_BORROW_REPAY_HISTORY, params)

    # POST /api/v5/account/fixed-loan/convert-to-market-loan
    def convert_to_market_loan(self,ordId = ''):
        params = {'ordId':ordId}
        return self._request_with_params(POST, CONVERT_TO_MARKET_LOAN, params)

    # POST /api/v5/account/fixed-loan/reduce-liabilities
    def reduce_liabilities(self,ordId = '',pendingRepay = ''):
        params = {'ordId':ordId,'pendingRepay':pendingRepay}
        return self._request_with_params(POST, REDYCE_LIABILITIES, params)
        
    # GET /api/v5/trade/account-rate-limit
    def account_rate_limit(self,):
        params = {}
        return self._request_with_params(GET, ACC_RATE_LIMIT, params)

    # POST /api/v5/account/bills-history-archive
    def bills_history_archive(self,year = '', quarter = ''):
        params = {'year':year, 'quarter':quarter}
        return self._request_with_params(POST, BILLS_HISTORY_ARCHIVE, params)

    # GET /api/v5/account/bills-history-archive
    def get_bills_history_archive(self, year = '', quarter = ''):
        params = {'year':year, 'quarter':quarter}
        return self._request_with_params(GET, GET_BILLS_HISTORY_ARCHIVE, params)

